lib/audit_help.c

```diff
@@ -63,8 +82,16 @@ void audit_logger (int type, MAYBE_UNUSED const char *pgname, const char *op,
 	if (audit_fd < 0) {
 		return;
 	} else {
-		audit_log_acct_message (audit_fd, type, NULL, op, name, id,
-		                        NULL, NULL, NULL, result);
+		/*
+		 * The audit system needs white space in the op field to
+		 * be replaced with dashes so that parsers get the whole
+		 * field.
+		 */
+		char *fixed_op = strreplace (strdup (op), ' ', '-');
+		audit_log_acct_message (audit_fd, type, NULL,
+					fixed_op ? fixed_op : op, name,
```

Comment:
I don't think it's a good idea to fall back to `op` on allocation error.  This means that two formats can coexist in the same file, and software must expect both.

I think I would rather report the error to the caller (and maybe abort the program).

Reply:
If you make sure that `op` is always passed as a string literal, maybe you could use strdupa(3).

To make sure that an argument is a string literal, you could use this code:
<https://github.com/shadow-maint/shadow/pull/1189/files>
(we can merge it before this one, so that you can use it in your patch.)  Cc: @hallyn, @ikerexxe 

For now, you can cherry-pick that commit in your branch.

Then, you could define this:

```c
#define audit_logger(..., op, ...)  do \
{ \
        static_assert(is_string_literal(op)); \
        audit_logger_(..., strdupa(op), ...); \
} while (0)


void audit_logger_(...)
{...}
```